Sources/CreateAPI/Generator/Generator+Schemas.swift

```diff
@@ -349,6 +358,14 @@ extension Generator {
             }
         }
     }
+    
+    private func excludedProperties(for type: String) -> Set<String> {
+        let excluded = options.entities.exclude
+            .map { $0.split(separator: ".") }
+            .filter { $0.count == 2 && $0[0].elementsEqual(type) }
+            .map { String($0[1]) }
+        return Set(excluded)
```

Comment:
So this got me thinking a little about performance, but we don't really have any benchmarking in place and I can't see a significant problem here although I was trying to think of ways to pre-process this as much as possible when deserialising the config. 

I then also remembered the current implementation for entity name templating:

https://github.com/CreateAPI/CreateAPI/blob/83a871b63126ebf1978f28a13a026bdac2a4a2e9/Sources/CreateAPI/Generate.swift#L146-L149

As it stands, this would break because it assumes that the elements inside `exclude` are specific entity names and doesn't expect the dot notation for properties too. 

With that, I had an idea that maybe we can use a `RawRepresentatble` backed type instead of a string for this config option:

```swift 
public struct EntityExclude: RawRepresentable, Codable, Hashable {
    public var name: String
    public var property: String?

    public init(name: String, property: String?) {
        self.name = name
        self.property = property
    }

    public init?(rawValue: String) {
        let components = rawValue.components(separatedBy: ".")
        switch components.count {
        case 1:
            self.init(name: components[0], property: nil)
        case 2:
            self.init(name: components[0], property: components[1])
        default:
            return nil
        }
    }

    public var rawValue: String {
        if let property = property {
            return "\(name).\(property)"
        } else {
            return name
        }
    }
}
```

```swift
public var exclude: Set<EntityExclude> = []
```

With that, you should be able to property apply the entity name template and it also saves a little processing in this lookup here.

Reply:
Good catch and I approve of the `RawRepresentable` for pre-processing. I can take a look at this.